### PR TITLE
document-reference-update

### PR DESCRIPTION
**Pull request by teeohareeyeohin**
Description
----
Adding update information to document reference

Screen Shot 2020-09-10 at 3 04 58 PM

PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: content/millennium/r4/documents/document-reference.md [hunk 1]
```diff
@@ -160,6 +160,62 @@ X-Request-Id: 3e4cb2f732daacdb6cca2eb944e80e55
...
+## Update
+
+Update an existing document reference.
+
+    PUT /DocumentReference/:id
+
+_Implementation Notes_
+
+* Any field which is missing will be interpreted as nulling out or removing data from the resource. See [FHIR<sup>®</sup> Update] for additional details about update operations.
+* Both write and read scopes are required to update a document reference
```
Comment (mayur-rajendran) [inline content/millennium/r4/documents/document-reference.md:176]: we should no longer need read scopes for an update, we can remove this line.

File: lib/resources/r4/document_reference.yaml [hunk 2]
```diff
@@ -290,3 +291,42 @@ fields:
...
+
+- name: type
+  required: ‘Yes’
+  type: CodeableConcept
+  action:
+  - update
+  description: Specifies the particular kind of document referenced.
+  note: The type must include a LOINC or a proprietary coding but not both together. Multiple LOINC codings or a single proprietary coding can be provided.
+        <br/><br/>
+        When providing proprietary code system, it should be of format 'https://fhir.cerner.com/&lt;your EHR source id&gt;/codeSet/&lt;code set&gt;' (where code set is a Millennium code set). Example&#58; 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/72'.
```
Comment (mayur-rajendran) [inline lib/resources/r4/document_reference.yaml:316]: We're only allowing proprietary codes from codeset 72.  I think we can make that a little more clear here.

File: lib/resources/r4/document_reference.yaml [hunk 3]
```diff
@@ -290,3 +291,42 @@ fields:
...
       }
     }
+
+- name: docStatus
+  required: 'Yes'
+  type: code
+  action:
+    - update
+  description: The status of the underlying document.
+  note: Only `final` is supported for provider access. `final` and `amended` are supported for system access.
```
Comment (mayur-rajendran) [inline lib/resources/r4/document_reference.yaml:301]: We'll support both final and amended for all access types for updates.  We're also allowing 'entered-in-error' as a way to in-error documents.

File: lib/resources/r4/document_reference.yaml (hunk 2 above)
Comment (teeohareeyeohin) [reply lib/resources/r4/document_reference.yaml:316]: For consistency, I thought I'd make it like create's type. Should we change both?

Comment (mayur-rajendran) [reply lib/resources/r4/document_reference.yaml:316]: Yup, if it makes it seem like we could support multiple codeSets, we should.... a simple change might be to just do this?
```suggestion
        When providing proprietary code system, it should be of format 'https://fhir.cerner.com/&lt;your EHR source id&gt;/codeSet/&lt;code set&gt;' (where code set is a Millennium code set 72). Example&#58; 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/72'.
```

File: lib/resources/r4/document_reference.yaml (hunk 3 above)
Comment (teeohareeyeohin) [reply lib/resources/r4/document_reference.yaml]: You can find the changes here: https://github.com/cerner/fhir.cerner.com/pull/463/commits/2697a167ece46ac0a9e9b0a2a671208d4ce9715d

File: lib/resources/r4/document_reference.yaml (hunk 2 above)
Comment (teeohareeyeohin) [reply lib/resources/r4/document_reference.yaml]: You can find the changes here: https://github.com/cerner/fhir.cerner.com/pull/463/commits/2697a167ece46ac0a9e9b0a2a671208d4ce9715d

File: content/millennium/r4/documents/document-reference.md (hunk 1 above)
Comment (teeohareeyeohin) [reply content/millennium/r4/documents/document-reference.md]: You can find the changes here: https://github.com/cerner/fhir.cerner.com/pull/463/commits/2697a167ece46ac0a9e9b0a2a671208d4ce9715d

**Comment by MaxPhilips** [comment]
@teeohareeyeohin looks like adding the image in between the checkboxes caused the second box to be rendered incorrectly. I noticed this while checking current status of deployable PRs and noticed only 1 task was noted on this PR. I updated the description so the second task is tracked now

File: lib/resources/r4/document_reference.yaml [hunk 4]
```diff
@@ -212,6 +212,12 @@ fields:
    {
      "data": "PGh0bWw+Cjx0aXRsZT4gVGVzdCBEb2N1bWVudCA8L3RpdGxlPgoKRG9jdW1lbnQgY29udGVudCEKCjwvaHRtbD4="
    }
+  note: >
+    You can validate your document using any available strict XHTML 1.0 validator (eg: <a href="http://validator.w3.org/#validate_by_upload+with_options">w3 validator</a>
```
Comment (mayur-rajendran) [inline lib/resources/r4/document_reference.yaml:216]: Can you add something like: "If providing XHTML or HTML content, " to indicate that the validations and sanitization will only be run on those particular content types.

File: lib/resources/r4/document_reference.yaml [hunk 5]
```diff
@@ -290,3 +296,54 @@ fields:
...
       }
     }
+
+- name: docStatus
+  required: 'Yes'
+  type: code
+  action:
+    - update
+  description: The status of the underlying document.
+  note: Each status are supported for all access types&#58; `final`, `amended`, and `entered-in-error`. If either status or docStatus are updated to "entered-in-error", the document will be treated as a document that was entered in error, regardless of the values of the other fields.
```
Comment (mayur-rajendran) [inline lib/resources/r4/document_reference.yaml:306]: You can probably change this to something more like: 
```suggestion
  note: Allowed values for docStatus are&#58; `final`, `amended`, and `entered-in-error`. If either status or docStatus are updated to "entered-in-error", the document will be treated as a document that was entered in error, regardless of the values of the other fields.
```

File: lib/resources/r4/document_reference.yaml [hunk 6]
```diff
@@ -230,6 +243,7 @@ fields:
   type: dateTime
   action:
     - create
+    - update
   description: The date that the attachment was first created.
   url: https://www.hl7.org/fhir/datatypes-definitions.html#Attachment.creation
   note: A time component must be provided with the date.
```
Comment (mayur-rajendran) [inline lib/resources/r4/document_reference.yaml:249]: Can we add a note here about how the new creation date must not be earlier than the previous creation date.  We'll need to have a separate section for writes and updates.  Under the note section for the update action, we'll need to add:

```
  note: A time component must be provided with the date.  The new date must not be earlier than the previous content.attachment.creation date.
```

File: lib/resources/r4/document_reference.yaml [hunk 7]
```diff
@@ -59,7 +59,7 @@ fields:
   description: Specifies the particular kind of document referenced.
   note: The type must include a LOINC or a proprietary coding but not both together. Multiple LOINC codings or a single proprietary coding can be provided.
         <br/><br/>
-        When providing proprietary code system, it should be of format 'https://fhir.cerner.com/&lt;your EHR source id&gt;/codeSet/&lt;code set&gt;' (where code set is a Millennium code set). Example&#58; 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/72'.
+        When providing proprietary code system, it should be of format 'https://fhir.cerner.com/&lt;your EHR source id&gt;/codeSet/&lt;code set&gt;' (where code set is a Millennium code set 72). Example&#58; 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/72'.
```
Comment (DanielWhiteCerner) [inline lib/resources/r4/document_reference.yaml:62]: This shouldn't be "is a Millennium codeset 72". That doesn't make sense. "is a Millennium codeset, 72" or "is Millennium codeset 72" would both work instead.

File: lib/resources/r4/document_reference.yaml [hunk 8]
```diff
@@ -290,3 +305,99 @@ fields:
...
+
+- name: type
+  required: ‘Yes’
+  type: CodeableConcept
+  action:
+    - update
+  description: Specifies the particular kind of document referenced.
+  note: The type must include a LOINC or a proprietary coding but not both together. Multiple LOINC codings or a single proprietary coding can be provided.
+        <br/><br/>
+        When providing proprietary code system, it should be of format 'https://fhir.cerner.com/&lt;your EHR source id&gt;/codeSet/&lt;code set&gt;' (where code set is a Millennium code set 72). Example&#58; 'https://fhir.cerner.com/ec2458f2-1e24-41c8-b71b-0e701af7583d/codeSet/72'.
```
Comment (DanielWhiteCerner) [inline lib/resources/r4/document_reference.yaml:330]: Same codeset note here. Make sure we update it everywhere.

**Review by DanielWhiteCerner — APPROVED** [review]

File: lib/resources/r4/document_reference.yaml [hunk 9]
```diff
@@ -206,18 +211,26 @@ fields:
...
   action:
     - create
+    - update
   description: The actual data of the attachment - a sequence of bytes, base64 encoded.
   url: https://www.hl7.org/fhir/datatypes-definitions.html#Attachment.data
   example: |
    {
      "data": "PGh0bWw+Cjx0aXRsZT4gVGVzdCBEb2N1bWVudCA8L3RpdGxlPgoKRG9jdW1lbnQgY29udGVudCEKCjwvaHRtbD4="
    }
+  note: >
```
[the remark this answers is not in the document]
Comment (mayur-rajendran) [reply lib/resources/r4/document_reference.yaml:215]: nvm looks like its needed

**Review by mayur-rajendran — APPROVED** [review]

**Comment by ebaranowski** [comment]
Validated on live

**Comment by ebaranowski** [comment]
Screen Shot 2020-11-12 at 5 17 41 PM

Validated on live

**Pull request merged**